From 85dbca4d5dd6f1049f9146e0c9b5ccef927be87c Mon Sep 17 00:00:00 2001
From: Huynh Thanh Hung <hung.huynh.wz@renesas.com>
Date: Tue, 4 Aug 2020 09:48:52 +0700
Subject: [PATCH] arm64: dts: hihope-rzg2m: enable IPMMU for GSX

This patch will enable support IPMMU and IPMMU domain of GSX for:
* R8A774A1 HiHope RZG2M rev 4
* R8A774A1 HiHope RZG2M rev 2
* R8A774A3 HiHope RZG2M rev 4

Signed-off-by: Huynh Thanh Hung <hung.huynh.wz@renesas.com>
---
 arch/arm64/boot/dts/renesas/r8a774a1-hihope-rzg2m-rev2.dts | 13 +++++++++++++
 arch/arm64/boot/dts/renesas/r8a774a1-hihope-rzg2m.dts      | 13 +++++++++++++
 arch/arm64/boot/dts/renesas/r8a774a3-hihope-rzg2m.dts      | 12 ++++++++++++
 3 files changed, 38 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a774a1-hihope-rzg2m-rev2.dts b/arch/arm64/boot/dts/renesas/r8a774a1-hihope-rzg2m-rev2.dts
index 92b65aa..ffc5a69 100644
--- a/arch/arm64/boot/dts/renesas/r8a774a1-hihope-rzg2m-rev2.dts
+++ b/arch/arm64/boot/dts/renesas/r8a774a1-hihope-rzg2m-rev2.dts
@@ -88,3 +88,16 @@
 	clock-names = "du.0", "du.1", "du.2",
 		      "dclkin.0", "dclkin.1", "dclkin.2";
 };
+
+&ipmmu_mm {
+	status = "okay";
+};
+
+&ipmmu_pv0 {
+	status = "okay";
+};
+
+&ipmmu_pv1 {
+	status = "okay";
+};
+
diff --git a/arch/arm64/boot/dts/renesas/r8a774a1-hihope-rzg2m.dts b/arch/arm64/boot/dts/renesas/r8a774a1-hihope-rzg2m.dts
index 059e48f..4b8c355 100644
--- a/arch/arm64/boot/dts/renesas/r8a774a1-hihope-rzg2m.dts
+++ b/arch/arm64/boot/dts/renesas/r8a774a1-hihope-rzg2m.dts
@@ -88,3 +88,16 @@
 	clock-names = "du.0", "du.1", "du.2",
 		      "dclkin.0", "dclkin.1", "dclkin.2";
 };
+
+&ipmmu_mm {
+	status = "okay";
+};
+
+&ipmmu_pv0 {
+	status = "okay";
+};
+
+&ipmmu_pv1 {
+	status = "okay";
+};
+
diff --git a/arch/arm64/boot/dts/renesas/r8a774a3-hihope-rzg2m.dts b/arch/arm64/boot/dts/renesas/r8a774a3-hihope-rzg2m.dts
index 8343efd..467a3d9 100644
--- a/arch/arm64/boot/dts/renesas/r8a774a3-hihope-rzg2m.dts
+++ b/arch/arm64/boot/dts/renesas/r8a774a3-hihope-rzg2m.dts
@@ -98,3 +98,15 @@
 &vspi0 {
 	status = "okay";
 };
+
+&ipmmu_mm {
+	status = "okay";
+};
+
+&ipmmu_pv0 {
+	status = "okay";
+};
+
+&ipmmu_pv1 {
+	status = "okay";
+};
-- 
1.9.1

